packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -238,9 +240,18 @@ void dispose() {
   }
 
   public static void registerWith(Registrar registrar) {
+    final VideoPlayerPlugin plugin = new VideoPlayerPlugin(registrar);
     final MethodChannel channel =
         new MethodChannel(registrar.messenger(), "flutter.io/videoPlayer");
-    channel.setMethodCallHandler(new VideoPlayerPlugin(registrar));
+    channel.setMethodCallHandler(plugin);
+    registrar.addViewDestroyListener(
```

Comment:
I have not written java lambda's much but I think this can be shortened to:

```java
registrar.addViewDestroyListener((FlutterNativeView view) {
  plugin.onDestroy();
  return false;
});
```

or some such.

Reply:
I wrote it that way, but the PR checker complained that this is not compatible with Java 7 (I don't know why it cares about Java 7 ;).